sdk/src/sharedTest/java/io/customer/sdk/repository/ProfileRepositoryTest.kt

```diff
@@ -92,6 +94,29 @@ class ProfileRepositoryTest : BaseTest() {
         verifyNoMoreInteractions(backgroundQueueMock)
     }
 
+    @Test
+    fun identify_givenFirstTimeIdentify_givenNoDeviceTokenRegistered_expectProfileIdentifiedHookUpdateWithCorrectIdentifier() {
+        val givenIdentifier = String.random
+        val newIdentifier = String.random
+        val givenAttributes = mapOf("name" to String.random)
+        prefRepository.saveIdentifier(givenIdentifier)
+        whenever(
+            backgroundQueueMock.queueIdentifyProfile(
+                anyOrNull(),
+                anyOrNull(),
+                anyOrNull()
+            )
+        ).thenReturn(QueueModifyResult(true, QueueStatus(siteId, 1)))
+
+        repository.identify(newIdentifier, givenAttributes)
+
+        val argumentCaptor = nullableArgumentCaptor<ModuleHook.ProfileIdentifiedHook>()
+
+        verify(hooksManager, times(1)).onHookUpdate(argumentCaptor.capture())
+
+        assertEquals(newIdentifier, argumentCaptor.firstValue?.identifier)
```

Comment:
I have found that `.apply{}` works really nicely with argument captor. I find it's easier to write and read. I am OK with using it or not. 

```suggestion
        nullableArgumentCaptor<ModuleHook.ProfileIdentifiedHook>().apply {
            verify(hooksManager, times(1)).onHookUpdate(capture())
            assertEquals(newIdentifier, firstValue?.identifier)
        }
```